src/concrete/ml/sklearn/linear_model.py

```diff
@@ -418,90 +422,115 @@ def _fit_encrypted(
+            self.label_encoder = LabelEncoder()
+            self.label_encoder.fit(y)
+            self.classes_ = self.label_encoder.classes_
 
-        y = self.label_encoder.transform(y)
+            assert isinstance(self.classes_, numpy.ndarray)
 
-        # Get the inputs' extreme values
-        x_min, x_max = X.min(axis=0), X.max(axis=0)
+            if len(self.classes_) != 2:
+                raise NotImplementedError(
+                    f"Only binary classification is currently supported when FHE training is "
+                    f"enabled. Got {len(self.classes_)} labels: {self.classes_}."
+                )
 
-        # Build and compile the training quantized module
-        self.training_quantized_module = self._get_training_quantized_module(
-            x_min=x_min,
-            x_max=x_max,
-        )
+            # Get the inputs' extreme values
+            x_min, x_max = X.min(axis=0), X.max(axis=0)
 
-        if fhe == "execute":  # pragma: no cover
-            # Key generation
-            if self.verbose:
-                print("Key Generation...")
+            # Build and compile the training quantized module
+            self.training_quantized_module = self._get_training_quantized_module(
+                x_min=x_min,
+                x_max=x_max,
+            )
 
-            # mypy
+        y = self.label_encoder.transform(y)
+
+        # Key generation
+        if fhe == "execute":  # pragma: no cover
             assert self.training_quantized_module.fhe_circuit is not None
 
-            start = time.time()
-            self.training_quantized_module.fhe_circuit.keygen(force=False)
-            end = time.time()
-            if self.verbose:
-                print(f"Key generation took {end - start:.4f} seconds.")
+            # Generate the keys only if necessary. This is already done using the `force=False`
+            # parameter, but here we also avoid printing too much verbose if activated
+            if not self.training_quantized_module.fhe_circuit.keys.are_generated:
+                if self.verbose:
+                    print("Key Generation...")
+
+                start = time.time()
+                self.training_quantized_module.fhe_circuit.keygen(force=False)
+                end = time.time()
+
+                if self.verbose:
+                    print(f"Key generation took {end - start:.4f} seconds.")
 
         # Mypy
         assert self.parameters_range is not None
 
-        # Initialize the weight values
+        # Initialize the weight values with the given ones if some are provided
         if coef_init is not None:
             weights = coef_init
-        elif self.warm_start and self._weights_encrypted_fit is not None:
+
+        # Else, if warm start is activated or this is a partial fit, use some already computed
+        # weight values have if there are some
+        elif (self.warm_start or is_partial_fit) and self._weights_encrypted_fit is not None:
```

Comment:
this is more or less a fix : before, we were asking users to set `warm_start` to True in order to benefit from `partial_fit`, but that's not the case in sklearn (https://scikit-learn.org/stable/glossary.html#term-partial_fit)

Reply:
Good catch!